packages/realm-network-transport/src/DefaultNetworkTransport.ts

```diff
@@ -36,6 +37,29 @@ export class DefaultNetworkTransport implements NetworkTransport {
     "Content-Type": "application/json",
   };
 
+  private static createTimeoutSignal(timeoutMs: number | undefined) {
+    if (typeof timeoutMs === "number") {
+      const controller = new DefaultNetworkTransport.AbortController();
+      // Call abort after a specific number of milliseconds
+      const timeout = setTimeout(() => {
+        controller.abort();
+      }, timeoutMs);
+      return {
+        signal: controller.signal,
+        cancelTimeout: () => {
+          clearTimeout(timeout);
+        },
+      };
+    } else {
+      return {
+        signal: undefined,
+        cancelTimeout: () => {
+          /* No-op */
+        },
+      };
+    }
+  }
+
```

Comment:
I just moved this up to be a `static` member, since it didn't need access to the instance.